File: web/nexus-ui/src/contracts/requests/register-user.request.ts
export interface RegisterRequest {
    login: string;
    userName: string;
    verifier: string;
    clientSalt: string;
    encryptedDek: string;
    cryptoVersion: number,
    srpVersion: number,
    encryptedVerifierWrapKey: string, 
    keyWrapVersion: number, 
    asymmetricKeyId: string,
    email: string;
    idGender: string | null;
    idCountry: string | null;
    recoveryKeys: Array<{encryptedValue: string, cryptoVersion: number}>
}

File: backend/src/shared/contracts/Shared.Contracts/RegisterUserRequest.cs
namespace Shared.Contracts
{
    public record RegisterUserRequest(
        string Login, string UserName, 
        string Verifier, int SrpVersion, string ClientSalt, 
        string EncryptedVerifierWrapKey, int CryptoVersion,
        string EncryptedDek, int KeyWrapVersion, string AsymmetricKeyId, 
        string Email, string? IdGender, string? IdCountry,
        IReadOnlyCollection<RecoveryKeyData> RecoveryKeys);

    public record RecoveryKeyData(string EncryptedValue, int CryptoVersion);
}